### PR TITLE
feat: unified programmatic rewards

### PR DESCRIPTION
**Pull request by 0xClandestine**

### CODE REVIEW

File: src/contracts/interfaces/IRewardsCoordinator.sol [hunk 1]
```diff
@@ -169,6 +169,23 @@ interface IRewardsCoordinator {
...
+    /// @param amount The total amount of tokens transferred to the RewardsCoordinator.
+    /// @param contentHash The keccak256 hash of the JSON manifest file listing the earners and amounts.
+    /// @param contentURI The publicly addressable URI pointing to the JSON manifest file.
+    event DirectRewardPayment(
+        address indexed sender,
+        address indexed avs,
+        IERC20 token,
+        uint256 amount,
+        bytes32 contentHash,
+        string contentURI
```
Comment (ypatil12) [inline src/contracts/interfaces/IRewardsCoordinator.sol:186]: We should index the contentURI IMO

**Review by ypatil12 — COMMENTED** [review]
Can we update this to be off of dev and go into a new `feat/uniprint` release branch?

File: src/contracts/interfaces/IRewardsCoordinator.sol (hunk 1 above)
Comment (0xClandestine) [reply src/contracts/interfaces/IRewardsCoordinator.sol:186]: Can dynamic types be indexed?

Comment (ypatil12) [reply src/contracts/interfaces/IRewardsCoordinator.sol:186]: Ahh good point, ya we can't

**Comment by ypatil12** [comment]
Everything up to https://github.com/Layr-Labs/eigenlayer-contracts/pull/701/commits/f35aedff7c4345268ff243480b883dca61c54204 LGTM

**Comment by 0xClandestine** [comment]
TODO: Modify `submitClaim` such that it takes in an array to allow for multiple claims.

**Comment by ypatil12** [comment]
Closing in favor of https://github.com/Layr-Labs/eigenlayer-contracts/pull/837

**Pull request closed without merging**